[CI] Increases the number of lit workers based on the GPU arch.

stefankoncarevic opened the pull request:
- For MFMA architectures (gfx908, gfx90a, gfx942, gfx950), the worker count is increased to 64.
- For Navi architectures, the worker count is increased to 16.

The goal is to evaluate how CI machines handle more threads, the tests executions remain stable under these settings. This change is an initial step to experiment with performance and stability improvements during CI runs.

stefankoncarevic committed:
[CI] Increases the number of lit workers based on the GPU arch.

- For MFMA architectures (gfx908, gfx90a, gfx942, gfx950),
  the worker count is increased to 64.
- For Navi architectures, the worker count is increased to 16.

The goal is to evaluate how CI machines handle more threads,
the tests executions remain stable under these settings.
This change is an initial step to experiment with performance
and stability improvements during CI runs.

dhernandez0 commented on mlir/utils/jenkins/Jenkinsfile:
```diff
@@ -353,10 +353,11 @@ def rebootNode() {
     def gpu_arch = get_gpu_architecture()
-    if (gpu_arch.contains('gfx942') || gpu_arch.contains('gfx950')) {
-        limit_lit_workers = 40
+    def mfma_archs = ['gfx908', 'gfx90a', 'gfx942', 'gfx950']
```
nit: check if starts with gfx9

stefankoncarevic committed:
Merge branch 'develop' into lit-workers

stefankoncarevic committed:
Limit LIT worker count for unstable CI architectures.
- Reduce LIT worker count to 20 for gfx90a/gfx908 due to CI
instability and test timeouts.
- Set worker count to 8 for Navi3x/Navi4x due to instability
on some navi CI machines.

stefankoncarevic committed:
Merge branch 'develop' into lit-workers

stefankoncarevic committed:
Merge branch 'develop' into lit-workers

stefankoncarevic committed:
Temporarily disabled data-after-header validation from tuning
file check.

dorde-antic commented on mlir/utils/tuna/tuna-script.sh:
```diff
@@ -59,12 +59,13 @@ validate_tuning_file() {
-        exit 1
-    fi
+    #data_line=$(awk 'NR>1 && $0 !~ /^\s*$/ && $0 != header {print; exit}' header="$header" "$f")
+    #if [ -z "$data_line" ]; then
```
https://github.com/ROCm/rocMLIR-internal/issues/1950
